### DISCUSSION TITLE
［V23 Beta2］［终端］添加更改终端透明度的快捷键

### DISCUSSION THREAD

**Discussion by ArchieMeng** [linuxdeepin/developer-center#3843] [Features Request & Ideas | 特性请求 & 头脑风暴]
当前deepin-terminal没有修改背景透明度的快捷键，修改需要到设置中手动调整．

**Comment by xzl01** [comment]
我认为这个不应该作为一个快捷键，你可以让用户选择自定义的key-bind

**Reply by ArchieMeng** [reply]
就是说可以自定义的快捷键？

**Reply by xzl01** [reply]
是的 可以在设置里面提供选项

**Reply by ArchieMeng** [reply]
可以的，我就有这个意思．

**Reply by liwl1991** [reply]
@ArchieMeng 哎我去，今天刚提了issue,就看到要开发，这个挺不错的

**Reply by ArchieMeng** [reply]
@liwl1991 我看到你提的Issue了．你已经实现出来了是吗？可以试着Cherry-Pick相关Commit到主线分支，先看看有没有什么冲突吧．

**Reply by liwl1991** [reply]
@ArchieMeng 我这周三4.5提交一个pr看看吧

**Reply by liwl1991** [reply]
@ArchieMeng 我已经提交另一个pr，看看是否符合规范？

**Reply by ArchieMeng** [reply]
@liwl1991 感谢提交PR．其它都行，就是功能和我想的有点不一样．我设想的是两个快捷键分别实现加减终端透明度这样．如果只是切换的话，会不太自由．不过，这个也是可以一起讨论的．

**Comment by black-desk** [comment]
我觉得这个东西有点 新

好像就没听说过有什么终端支持用快捷键调透明度(x

不过我想了想好像确实有那么一点必要: 透明度其实和 下层窗口的颜色 有很强的关系. 不太可能用同一个透明度设置让用户用的时候觉得都很舒服.

所以其实除了给一个调节透明度的快捷键, 还有另一种可能的处理方式:

是不是可以考虑做一个 让用户自定义曲线, 根据下层窗口的亮度动态调节透明度的功能. 应用可以拿到下层窗口的透明度么? @justforlxz

**Reply by BLumia** [reply]
> 好像就没听说过有什么终端支持用快捷键调透明度(x

这个还确实是 feature，之前 gtk/vala 版就支持 <kbd>Ctrl+滚轮</kbd> 的方式调整透明度，设计初衷是用户可能在看一些文档资料了解一些命令怎么输入，然后终端窗口在资料的上方。通过这种方式临时条节透明度可以让用户不切换窗口即可看清终端后窗口的文字，然后进行输入，并在不再需要时调回去。然后这个需求在做初版 dtk 版终端的时候我没加，可能后面接手的人不知道这个需求（（

**Reply by black-desk** [reply]
> > 好像就没听说过有什么终端支持用快捷键调透明度(x
> 
> 这个还确实是 feature，之前 gtk/vala 版就支持 Ctrl+滚轮 的方式调整透明度，设计初衷是用户可能在看一些文档资料了解一些命令怎么输入，然后终端窗口在资料的上方。通过这种方式临时条节透明度可以让用户不切换窗口即可看清终端后窗口的文字，然后进行输入，并在不再需要时调回去。然后这个需求在做初版 dtk 版终端的时候我没加，可能后面接手的人不知道这个需求（（

那我觉得你这个(x 滚轮的设计就非常不vim

怎么能让人用终端的时候伸手去摸鼠标呢

**Reply by BLumia** [reply]
对，这个是我最初没加的原因（（（（

**Reply by black-desk** [reply]
仔细想了想这种功能似乎是弄成可配置性更高一些的样子会比较好 每个用户用起来的感觉可能都不太一样(x

**Reply by ArchieMeng** [reply]
谁用深度终端了还没有个鼠标啊喂．XD
不过，这个切换需求好像还有点道理．虽然我不会这么用终端的．

**Reply by BLumia** [reply]
其实最初 gtk 版选滚轮还是蛮可以理解的。因为用户其实不在乎这个透明度的具体数值，滚轮反倒可以方便用户快速的调整到一个舒适的值上。

不过就此需求而言，对于调高与调低透明度，我觉得给个自由设置快捷键的配置就可以了，哪怕默认为空。

**Reply by liwl1991** [reply]
xshell的透明度可以用快捷键的，设定一个初始的值，通过快捷键直接调整到这个透明度，当然也支持滑杆进行细颗粒度调整

**Reply by liwl1991** [reply]
应该切换透明和不透明，而非调整透明度的范围

**Reply by ArchieMeng** [reply]
@BLumia @black-desk 当前PR链接　https://github.com/linuxdeepin/deepin-terminal/pull/268

**Reply by ArchieMeng** [reply]
> xshell的透明度可以用快捷键的，设定一个初始的值，通过快捷键直接调整到这个透明度，当然也支持滑杆进行细颗粒度调整

设置成固定调整值会存在覆盖用户原有透明度设置的问题．所以我觉得还是给个快速调整而不是切换的会更好些．

**Reply by liwl1991** [reply]
是的，是有这个问题。
其实一开始我也是这么考虑的，类似xshell给个初始值的设置选项。
通过滑轮修改这个值，比如启用透明度时，直接设置终端透明度到这个值，然后可以通过滑轮修改这个值（修改后的值就是下一次的默认值）。而快捷键则提供不透明到透明度（当前设置值）的切换。
但我能力有限，并且这个对原有功能破坏很大（把原有的滑杆设置2个触发，一个是改值，一个是该透明度），就没去实现。

**Comment by meiyixiang** [comment]
我认为，可以给现有的透明图调节增加一个快捷键，但在全黑和一个固定的透明值之间切换，不太符合大多数用户的使用。可能有的用户觉得30%透明度更合适 ，有些觉得60%更合适。可以给透明图新增快捷键盘+鼠标滚轮支持快速的调节透明度，也可以满足更多人的诉求。

**Reply by liwl1991** [reply]
> 我认为，可以给现有的透明图调节增加一个快捷键，但在全黑和一个固定的透明值之间切换，不太符合大多数用户的使用。可能有的用户觉得30%透明度更合适 ，有些觉得60%更合适。可以给透明图新增快捷键盘+鼠标滚轮支持快速的调节透明度，也可以满足更多人的宿醉。

是的。其实可以模仿xshell的操作的：滑杆可以调节透明度的同时，调节透明度的值。快捷键就用于在不透明和这个透明读值之前切换。既满足切换的需求，也满足不同透明度的需求。一般来说，用户调整透明度的频次很低，他不会先调到60,再调到30这样，而是调到了60以后，希望在60和100之间切换。

**Comment by meiyixiang** [comment]
你在使用过程中，在透明和不透明的评率会很高吗？

**Reply by liwl1991** [reply]
> 你在使用过程中，在透明和不透明的评率会很高吗？

被你说对了，我是非常高的。比如我想在终端透视下，看教程博客，教程视频等。单显示器情况下。
如果你用atl+tab,或者双显示器，终端透明本身就是伪需求了

**Comment by meiyixiang** [comment]
如果不是我觉得，提供给你可以快捷调整透明度的方式就可以了，没有百分之百的快捷键。因为我感觉这种在透明与不透明之间切换就不算是是个高频的操作，你需要的是能够在不开启设置的情况下调节合适的窗口透明度。

**Reply by liwl1991** [reply]
> 如果不是我觉得，提供给你可以快捷调整透明度的方式就可以了，没有百分之百的快捷键。因为我感觉这种在透明与不透明之间切换就不算是是个高频的操作，你需要的是能够在不开启设置的情况下调节合适的窗口透明度。
嗯，这个主要看你们产品经理如何去决策了哈，我这个毕竟是小众需求，使用场景频率都很低

**Comment by meiyixiang** [comment]
因为我是产品，所以我更希望这个功能能被更多人接受，例如：提供ctrl+alt+滚轮给你你在使用时也是可以快速通过滚轮去调节窗口透明度的。
但对于其他用户而言，你所描述的做法，相当于在透明度处又给了用户一个选择，可以在两个透明度直接切换。假定你需要的是A状态透明度为40%B状态为全黑，有可能另一个用户期望的是A透明度为10% B状态为60%。其实这样的功能让人很难理解，且功能非常的用于。

**Reply by liwl1991** [reply]
> 因为我是产品，所以我更希望这个功能能被更多人接受，例如：提供ctrl+alt+滚轮给你你在使用时也是可以快速通过滚轮去调节窗口透明度的。 但对于其他用户而言，你所描述的做法，相当于在透明度处又给了用户一个选择，可以在两个透明度直接切换。假定你需要的是A状态透明度为40%B状态为全黑，有可能另一个用户期望的是A透明度为10% B状态为60%。其实这样的功能让人很难理解，且功能非常的用于。

是的，应该考虑更大众一些的需求。

**Comment by meiyixiang** [comment]
OK，不过感谢你的贡献和像我们提交的PR，虽然可能不会完全符合你的操作习惯，但是我们会尽可能的以你的建议去满足更多的用户。
